processing/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -52,44 +58,107 @@
  * It is used by Lookup configuration manager to add/remove or list lookups configuration via HTTP or other protocols.
  * It does periodic snap shot of the list of lookup in order to bootstrap nodes after restart.
  */
-
 @ManageLifecycle
 public class LookupReferencesManager
 {
-  private static final Logger LOGGER = new Logger(LookupReferencesManager.class);
-  private final ConcurrentMap<String, LookupExtractorFactory> lookupMap = new ConcurrentHashMap<>();
-  // This is a lock against the state of the REFERENCE MANAGER (aka start/stop state), NOT of the lookup itself.
+  private static final EmittingLogger LOG = new EmittingLogger(LookupReferencesManager.class);
+
   private final ReadWriteLock startStopLock = new ReentrantReadWriteLock(true);
-  private final AtomicBoolean started = new AtomicBoolean(false);
+
+  @VisibleForTesting
+  boolean started = false;
+
+  @GuardedBy("startStopLock")
+  private final Map<String, LookupExtractorFactoryContainer> lookupMap = new HashMap<>();
 
   private final LookupSnapshotTaker lookupSnapshotTaker;
 
+  @VisibleForTesting
+  final BlockingQueue<Notice> queue = new ArrayBlockingQueue<>(10000);
+
+  private volatile ExecutorService exec;
```

Comment:
Convention in other parts of druid is to use single threaded executors instead of a Thread. partly because the Execs class has a bunch of boilerplate stuff for thread handling to keep things consistent across the project.